src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/BQSRSpark.java

```diff
@@ -0,0 +1,102 @@
+        usageExample = "BQSRSpark -I in.bam --BQSRKnownVariants in.vcf -O out.bam",
+        programGroup = SparkProgramGroup.class
+)
+/**
+ * BQSR. The final result is analysis-ready reads.
+ */
+public final class BQSRSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public boolean requiresReads() { return true; }
+
+    @Override
+    public boolean requiresReference() { return true; }
+
+    @Argument(doc = "the known variants", shortName = "BQSRKnownVariants", fullName = "baseRecalibrationKnownVariants", optional = true)
+    protected List<String> baseRecalibrationKnownVariants;
+
+    @Argument(doc = "the output bam", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    protected String output;
+
+    @Argument(doc = "If specified, shard the output bam", shortName = "shardedOutput", fullName = "shardedOutput", optional = true)
+    private boolean shardedOutput = false;
+
+    @Argument(doc = "the join strategy for reference bases", shortName = "joinStrategy", fullName = "joinStrategy", optional = true)
+    private JoinStrategy joinStrategy = JoinStrategy.SHUFFLE;
+    
+    @Override
+    public SerializableFunction<GATKRead, SimpleInterval> getReferenceWindowFunction() {
+        return BaseRecalibrationEngine.BQSR_REFERENCE_WINDOW_FUNCTION;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        JavaRDD<GATKRead> markedReads = getReads();
+        VariantsSparkSource variantsSparkSource = new VariantsSparkSource(ctx);
+
+        if ( baseRecalibrationKnownVariants.isEmpty() ) { // Warn the user if no dbSNP file or other variant mask was specified
+            throw new UserException.CommandLineException(BaseRecalibrator.NO_DBSNP_EXCEPTION);
+        }
+
+        // TODO: workaround for known bug in List version of getParallelVariants
+        if ( baseRecalibrationKnownVariants.size() > 1 ) {
+            throw new GATKException("Cannot currently handle more than one known sites file, " +
+                                    "as getParallelVariants(List) is broken");
+        }
+        JavaRDD<Variant> bqsrKnownVariants = variantsSparkSource.getParallelVariants(baseRecalibrationKnownVariants.get(0));
+
+        // TODO: Look into broadcasting the reference to all of the workers. This would make AddContextDataToReadSpark
+        // TODO: and ApplyBQSRStub simpler (#855).
+        JavaPairRDD<GATKRead, ReadContextData> rddReadContext = AddContextDataToReadSpark.add(markedReads, getReference(), bqsrKnownVariants, joinStrategy);
+        // TODO: broadcast the reads header?
+
+        //note: we use the reference dictionary from the reads themselves.
+        final RecalibrationReport bqsrReport = BaseRecalibratorSparkFn.apply(rddReadContext, getHeaderForReads(), getHeaderForReads().getSequenceDictionary(), new RecalibrationArgumentCollection());
```

Comment:
Shouldn't you be exposing the `RecalibrationArgumentCollection` as an `@ArgumentCollection` here (as in `BaseRecalibratorSpark`) so that the BQSR settings can be customized?